webrtc/libjingle/xmpp/boshsocket.h

```diff
@@ -0,0 +1,109 @@
+#ifndef WEBRTC_LIBJINGLE_XMPP_BOSHSOCKET_H_
+#define WEBRTC_LIBJINGLE_XMPP_BOSHSOCKET_H_
+
+#include "webrtc/libjingle/xmpp/asyncsocket.h"
+#include "webrtc/libjingle/xmpp/xmppengine.h"
+#include "webrtc/base/asyncsocket.h"
+#include "webrtc/base/bytebuffer.h"
+#include "webrtc/base/sigslot.h"
+#include <queue>
+
+namespace rtc {
+  class StreamInterface;
+  class SocketAddress;
+}
+
+namespace buzz {
+  typedef struct ManagedSocket{
+    rtc::AsyncSocket *socket_;
+#ifdef USE_SSLSTREAM
+    rtc::StreamInterface *stream_;
+#endif // USE_SSLSTREAM
+    bool pending_;
+    bool connected_;
+    bool connect_sent_;  // CONNECT sent, required when use ssl under proxy server.
+    buzz::AsyncSocket::State state_;
+
+    ManagedSocket() {
+      socket_ = NULL;
+#ifdef USE_SSLSTREAM
+      stream_ = NULL;
+#endif // USE_SSLSTREAM
+      pending_ = true;
+      connected_ = false;
+      connect_sent_ = false;
+      state_ = buzz::AsyncSocket::State::STATE_CLOSED;
+    }
+  } ManagedSocket;
+
+  typedef struct OutputMsg{
+    char *data_;
+    size_t len_;
+    OutputMsg(const char *str, size_t len) {
+      data_ = new char[len];
+      strcpy(data_, str);
```

Comment:
if the len is accurate, memcpy might be better -- strcpy() has some security implications. Also should make sure we have guards around so errant signaling can't create buffer issues.